c2corg_ui/externs/appx.js

```diff
@@ -84,3 +84,186 @@ appx.BootstrapSliderValues;
+ *   
+ *   height_diff_up: number,
+ *   height_diff_down: number,
+ *   height_diff_access: number,
+ *   height_diff_difficulties: number,
+ *   lift_access: boolean,
+ *   elevation_min : number,
+ *   elevation_max : number,
+ *   route_type: Array<string>,
+ *   orientations: Array<string>,
+ *   durations: number,
+ *   main_waypoint_id: number,
+ *   activities: Array<Object>,
+ *   route_length: number,
+ *   configuration: Array<string>,
+ *   ski_rating: string,
+ *   ski_exposition: string,
+ *   labande_ski_rating: string,
+ *   labande_global_rating: string,
+ *   difficulties_height: number,
+ *   engagement_rating: string,
+ *   equipment_rating: string,
+ *   global_rating: string,
+ *   ice_rating: string,
+ *   geometry: Object,
+ *   risk_rating: string,
+ *   rock_types: Array<string>,
+ *   mixed_rating: string,
+ *   glacier_gear: string,
+ *   exposition_rock_rating: string,
+ *   rock_free_rating: string,
+ *   rock_required_rating: string,
+ *   aid_rating: string,
+ *   climbing_outdoor_type: string,
+ *   hiking_mtb_exposition: string,
+ *   hiking_rating: string,
+ *   snowshoe_rating: string,
+ *   mtb_up_rating: string,
+ *   mtb_down_rating: string,
+ *   mtb_length_asphalt: string,
+ *   mtb_length_trail: string,
+ *   mtb_height_diff_portages: string,
+ *   via_ferrata_rating: string
+ * }}
+ */
+appx.Route;
+
+/**
+ * @typedef {{
+ *   associations: appx.DocumentAssociations,
+ *   locales: Array<Object>,
+ *   type : string,
+ *   quality: string,
+ *   document_id: number,
+ *   
+ *   elevation: number,
+ *   elevation_min: number,
+ *   maps: Array<Object>,
+ *   maps_info: string,
+ *   areas: Array<Object>,
+ *   public_transportation_types: Array<string>,
+ *   public_transportation_rating: string,
+ *   prominence: number,
+ *   height_max: number,
+ *   height_min: number,
+ *   height_median: number,
+ *   length: number,
+ *   geometry: Object,
+ *   slope: (number | string),
+ *   ground_types: Array<string>,
+ *   paragliding_rating: string,
+ *   exposition_rating: string,
+ *   routes_quantity: number,
+ *   rain_proof: string,
+ *   children_proof: string,
+ *   access_time: string,
+ *   snow_clearance_rating: string,
+ *   lift_access: (boolean | null),
+ *   parking_fee: (boolean | string),
+ *   equipment_ratings: Array<string>,
+ *   climbing_rating_min: string,
+ *   climbing_rating_median: string,
+ *   climbing_rating_max: string,
+ *   url: string,
+ *   climbing_styles: Array<string>,
+ *   orientations: Array<string>,
+ *   equipment_rating: string,
+ *   best_periods: Array<string>,
+ *   climbing_outdoor_types: Array<string>,
+ *   phone: (number | string),
+ *   capacity: number,
+ *   capacity_staffed: number,
+ *   phone_custodian: (number | string),
+ *   custodianship: string,
+ *   matress_unstaffed: (string | boolean),
+ *   blanket_unstaffed: (string | boolean),
+ *   gas_unstaffed: (string | boolean),
+ *   heating_unstaffed: (string | boolean)
+ * }}
+ */
+appx.Waypoint;
```

Comment:
@asaunier, if we are sure we will never have a use of some properties it is possible to omit them.
However if we set a non declared property there will be no warning and it will be minified.
Don't know what is worst.